python/ray/serve/_private/autoscaling_state.py

```diff
@@ -102,8 +102,12 @@ def __init__(self, deployment_id: DeploymentID):
 
     def register(
         self, info: DeploymentInfo, curr_target_num_replicas: Optional[int] = None
-    ) -> int:
-        """Registers an autoscaling deployment's info."""
+    ) -> Optional[int]:
```

Comment:
Do the callers of `register` handle a `None` return type gracefully?

Reply:
Yeah only `deploy` uses the return value of `register_deployment() -> register()`, and that callsite is guaranteed to pass in a non-null num replicas

Reply:
Could you update the return type of [`register_deployment`](https://github.com/ray-project/ray/blob/9a3b22b566119b6c2a1ce479c843e6a4414fd9e5/python/ray/serve/_private/autoscaling_state.py#L330-L335) since it can return `None` now?

Reply:
Updated!

Reply:
per offline discussion, also added an assertion to make it clear register should never return None in `deploy()`

Reply:
Thanks!